File: Source/Aladdin.ASN1.ISO/ASN1/ISO/CertificateSet.cs
using System;
using System.Runtime.Serialization;

// CertificateSet ::= SET OF CertificateChoices

namespace Aladdin.ASN1.ISO
{
	[Serializable]
	public class CertificateSet : Set
	{
		// конструктор при сериализации
        protected CertificateSet(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public CertificateSet(IEncodable encodable) : 
			base(new ChoiceCreator<CertificateChoices>().Factory(), encodable) {} 

		// конструктор при закодировании
		public CertificateSet(params IEncodable[] values) : 
			base(new ChoiceCreator<CertificateChoices>().Factory(), values) {} 
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/CertificateSet.java
package aladdin.asn1.iso;
import aladdin.asn1.*; 
import java.io.*;

// CertificateSet ::= SET OF CertificateChoices

public final class CertificateSet extends Set<IEncodable>
{
    private static final long serialVersionUID = 727826057387120526L;
    
	// конструктор при раскодировании
	public CertificateSet(IEncodable encodable) throws IOException 
	{
		super(new ChoiceCreator(CertificateChoices.class).factory(), encodable); 
	} 
	// конструктор при закодировании
	public CertificateSet(IEncodable... values) 
	{
		super(new ChoiceCreator(CertificateChoices.class).factory(), values); 	
	} 
}

